cmd/platform/export.go

```diff
@@ -1,39 +1,205 @@
-	This application is a tool to generate the needed files
-	to quickly create a Cobra application.`,
-		Run: func(cmd *cobra.Command, args []string) {
-			// Just use the no-op connector for now by default
-			exportableConnectors := []connector.Exportable{
-				noop.Connector(),
-			}
-			//TODO selectable format and output location
-			err := exportableConnectors[0].Export(connector.ENUMEXPORTFORMAT_HCL, "/tmp")
+		Short: "",
+		Long:  ``,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			l := logger.Get()
+
+			l.Debug().Msgf("Export Subcommand Called.")
+
+			apiClient, err := initApiClient(cmd.Context(), cmd)
 			if err != nil {
 				output.Format(cmd, output.CommandOutput{
-					Message: "Export failed.",
-					Fatal:   err,
+					Message: "Unable to initialize PingOne SDK client",
 					Result:  output.ENUMCOMMANDOUTPUTRESULT_FAILURE,
 				})
+				return err
 			}
+
+			// Using the --service parameter(s) provided by user, build list of connectors to export
+			exportableConnectors := []connector.Exportable{}
+			for _, service := range services {
+				switch service {
+				case pingone_platform.ServiceName:
+					exportableConnectors = append(exportableConnectors, pingone_platform.Connector(cmd.Context(), apiClient, viper.GetString(pingoneWorkerEnvironmentIdParamConfigKey)))
+				default:
+					output.Format(cmd, output.CommandOutput{
+						Message: fmt.Sprintf("Provided service not recognized: %s", service),
+						Result:  output.ENUMCOMMANDOUTPUTRESULT_FAILURE,
+					})
+					return err
+				}
+			}
+
+			// Select export format based on user's --export-format parameter
+			var connectorExportFormat string
+			switch exportFormat {
+			case connector.ENUMEXPORTFORMAT_HCL:
+				connectorExportFormat = connector.ENUMEXPORTFORMAT_HCL
+			default:
+				output.Format(cmd, output.CommandOutput{
+					Message: fmt.Sprintf("Provided export format not recognized: %s. Defaulting to HCL export format.", exportFormat),
+					Result:  output.ENUMCOMMANDOUTPUTRESULT_NOACTION_WARN,
+				})
+				connectorExportFormat = connector.ENUMEXPORTFORMAT_HCL
+			}
+
+			// Loop through user defined exportable connectors and export them
+			for _, connector := range exportableConnectors {
+				l.Debug().Msgf("Exporting %s service...", connector.ConnectorServiceName())
+
+				err := connector.Export(connectorExportFormat, outputDir, overwriteExport)
+				if err != nil {
+					output.Format(cmd, output.CommandOutput{
+						Message: fmt.Sprintf("Export failed for service: %s.", connector.ConnectorServiceName()),
+						Result:  output.ENUMCOMMANDOUTPUTRESULT_FAILURE,
+					})
+					return err
+				}
+			}
+			return nil
 		},
 	}
 
+	// Add flags that are not tracked in the viper configuration file
+	cmd.Flags().StringVar(&exportFormat, "export-format", "HCL", "Specifies export format\nValid options: 'HCL'")
+	cmd.Flags().StringSliceVar(&services, "service", []string{"pingone-platform"}, "Specifies service(s) to export")
```

Comment:
Does cobra automatically print the valid options here?

It looks like a decent amount of effort but in the future we may want to add a custom enum flag type for things like this - https://stackoverflow.com/questions/50824554/permitted-flag-values-for-cobra

Reply:
If gone ahead and added this work, since this PR is meant to set the framework for the remaining resources and services.